### PR TITLE
feat: Update Twtter Download Copy Nested buttons

### PR DESCRIPTION
**Pull request by Si-ege**

### CODE REVIEW

File: src/components/activity/index.js [hunk 1]
```diff
@@ -12,6 +12,7 @@ import { saveAs } from 'file-saver';
 import CustomizedDialogs, { MarkdownLink } from "../shareDialog";
 import { DEFAULT_COLOR, generateMonthlyActivityOption } from "../../constants";
 import { handleShareToTwitterClick } from "../../utils";
+import {DialogBox} from '../dialogBox'
```
Comment (guoqqqi) [inline src/components/activity/index.js:15]: format code

File: src/components/activity/index.js [hunk 2]
```diff
@@ -252,7 +253,7 @@ const ActivityChart = ({
         setLoading(false);
       });
   }, [repoList]);
-
+
```
Comment (guoqqqi) [inline src/components/activity/index.js:256]:
```suggestion

```

File: src/components/activity/index.js [hunk 3]
```diff
@@ -26,7 +27,7 @@ const ActivityChart = ({
   const [openAlert, setOpenAlert] = React.useState(false);
   const getShareParams = () =>
     `?chart=contributorMonthlyActivity&repo=${repoList.join(",")}`;
-  const [, setCopied] = useClipboard(`https://git-contributor.com/${getShareParams()}`, { successDuration: 3000 });
+  const [, setCopied] = useClipboard(`https://git-contributor.com/${getShareParams()}`,);
```
Comment (guoqqqi) [inline src/components/activity/index.js:30]: Can you explain what the reason for removing `successDuration` is?

File: src/components/activity/index.js [hunk 4]
```diff
@@ -294,6 +295,7 @@ const ActivityChart = ({
                         showLoading={loading}
                         notMerge
                       />
+                       <DialogBox  />
```
Comment (guoqqqi) [inline src/components/activity/index.js:298]:
```suggestion
                      <DialogBox  />
```

Comment (guoqqqi) [reply src/components/activity/index.js:298]: Please format your code before push

File: src/components/contributor/index.js [hunk 5]
```diff
@@ -428,6 +440,9 @@ const ContributorLineChart = ({
               showLoading={loading}
               notMerge
             />
+            
+              <DialogBox  />
+
```
Comment (guoqqqi) [inline src/components/contributor/index.js:445]:
```suggestion
              <DialogBox  />
```

File: src/components/dialogBox.jsx [hunk 6]
```diff
@@ -0,0 +1,204 @@
...
+      setEmOverTimecopy();
+      setOpenAlert(true);
+    }else{
+      setEmMonthlyActivecopy();
+      setOpenAlert(true);
+    }
+  }
+  
+  // const datasetWithFilters = [
+  //   ["ContributorNum", "Repo", "Date", "DateValue"],
```
Comment (guoqqqi) [inline src/components/dialogBox.jsx:89]: ?

File: src/components/dialogBox.jsx [hunk 7]
```diff
@@ -0,0 +1,204 @@
...
+import { Snackbar } from "@material-ui/core";
+import Alert from "@material-ui/lab/Alert";
+import DialogContentText from '@material-ui/core/DialogContentText';
+
+export const DialogBox = ({ repoList = ["apache/apisix"],}) => {
+  
+  const title = window.location.href.includes("contributorOverTime")
+    ? "contributor over time"
+    : "monthly active contributor"
+    console.log('shareText:',title)
```
Comment (guoqqqi) [inline src/components/dialogBox.jsx:22]: remove this `console`

File: src/components/dialogBox.jsx [hunk 8]
```diff
@@ -0,0 +1,204 @@
...
+  const Url = window.location.href.split('repo=apache/apisix')
+  const Tail = Url[1]
+  const OverTimegetShareParams = () => `?chart=contributorOverTime&repo=${repoList.join(",")}${Tail}`
+  const MonthlyActivegetShareParams = () => `?chart=contributorMonthlyActivity&repo=${repoList.join(",")}${Tail}`;
+  const OverTimevalue = `<iframe style={{ width: "100%", height: "auto" , minWidth: "600px", minHeight: "1000px" }} src="https://git-contributor.com/${OverTimegetShareParams()}" frameBorder="0"></iframe>`
+  const MonthlyActivevalue = `<iframe style={{ width: "100%", height: "auto" , minWidth: "600px", minHeight: "1000px" }} src="https://git-contributor.com/${MonthlyActivegetShareParams()}" frameBorder="0"></iframe>`
```
Comment (guoqqqi) [inline src/components/dialogBox.jsx:28]: We generally use small-hump nomenclature to declare variables, and large-hump nomenclature looks more like a component

File: src/components/dialogBox.jsx [hunk 9]
```diff
@@ -0,0 +1,204 @@
...
+  const [, setEmOverTimecopy] = useClipboard(OverTimevalue, { successDuration: 3000 })
+  const [, setEmMonthlyActivecopy] = useClipboard(MonthlyActivevalue, { successDuration: 3000 })
+  const [, setMonthlyActiveCopied] = useClipboard(`https://git-contributor.com/${MonthlyActivegetShareParams()}`, { successDuration: 3000 });
+  const [, setOverTimeCopied] = useClipboard(`https://git-contributor.com/${OverTimegetShareParams()}`, { successDuration: 3000 });
+  const [openAlert, setOpenAlert] = React.useState(false);
+  const [open, setOpen] = React.useState(false);
+  const [fullWidth, ] = React.useState(true);
+  const [maxWidth, ] = React.useState('sm');
```
Comment (guoqqqi) [inline src/components/dialogBox.jsx:36]: Do we really need so many states, can we reduce a few states?

File: src/components/dialogBox.jsx [hunk 10]
```diff
@@ -0,0 +1,204 @@
...
+    <Button
+      color="secondary"
+      variant="contained"
+      value="Copy"
+      style={{ padding: '5px 20px', textTransform: 'none', position: 'absolute', bottom: '10px', right: '20px',backgroundColor:' #E53E3E'}}
+      onClick={handleiframeiframe}
+    >
+      Copy
+    </Button>
+  );
```
Comment (guoqqqi) [inline src/components/dialogBox.jsx:49]:
```suggestion
    <Button
      color="secondary"
      variant="contained"
      value="Copy"
      style={{ padding: '5px 20px', textTransform: 'none', position: 'absolute', bottom: '10px', right: '20px',backgroundColor:' #E53E3E'}}
      onClick={handleiframeiframe}
    >
      Copy
    </Button>
  );
```
It's a good habit to format your code whenever you're done changing it

File: src/components/dialogBox.jsx [hunk 11]
```diff
@@ -0,0 +1,204 @@
...
+          Image
+        </Button>
+        <Button
+          color="warning"
+          variant="outlined"
+          size="small"
+          value="Embed"
+          startIcon={<CodeIcon />}
+          style={{ textTransform: 'none',marginRight: '10px'}}
+          onClick={handleClickOpen}
```
Comment (guoqqqi) [inline src/components/dialogBox.jsx:153]:
```suggestion
          onClick={() => setOpen(true)}
```

File: src/components/dialogBox.jsx [hunk 12]
```diff
@@ -0,0 +1,204 @@
...
+          onClick={handleShareClick}
+        >
+          Share on twtter
+        </Button>
+      </div>
+      <Dialog
+        fullWidth={fullWidth}
+        maxWidth={maxWidth}
+        open={open}
+        onClose={handleClose}
```
Comment (guoqqqi) [inline src/components/dialogBox.jsx:184]:
```suggestion
        onClose={() => setOpen(false)}
```

File: src/components/dialogBox.jsx [hunk 13]
```diff
@@ -0,0 +1,204 @@
...
+        aria-labelledby="responsive-dialog-title"
+      >
+        <DialogActions style={{ display: 'flex', justifyContent: 'space-between', borderBottom: '1px solid #c4c4c4' }}>
+          <DialogTitle size="small" style={{ padding: '10px' }}>Emeb Chart</DialogTitle>
+          <Button onClick={handleClose} color="warning" size="small" startIcon={<CloseIcon />} autoFocus>
```
Comment (guoqqqi) [inline src/components/dialogBox.jsx:189]:
```suggestion
          <Button onClick={() => setOpen(false)} color="warning" size="small" startIcon={<CloseIcon />} autoFocus>
```

**Comment by guoqqqi** [comment]
please check the `color` and the `font-size`
image

**Comment by guoqqqi** [comment]
image
Make the spacing smaller

**Comment by guoqqqi** [comment]
image
please add the `word-break: break-all`

File: src/components/DialogBox/index.jsx [hunk 14]
```diff
@@ -0,0 +1,155 @@
...
+import GetAppIcon from '@material-ui/icons/GetApp';
+import FilterNoneOutlinedIcon from '@material-ui/icons/FilterNoneOutlined';
+import useClipboard from "react-use-clipboard";
+import { handleShareToTwitterClick } from "../../utils";
+import { Snackbar, makeStyles } from "@material-ui/core";
+import Alert from "@material-ui/lab/Alert";
+import DialogContentText from '@material-ui/core/DialogContentText';
+
+export const DialogBox = ({ params = "" }) => {
+
```
Comment (guoqqqi) [inline src/components/DialogBox/index.jsx:18]:
```suggestion
```

File: src/components/DialogBox/index.jsx [hunk 15]
```diff
@@ -0,0 +1,155 @@
...
+      color: '#3e434a',
+      backgroundColor: '#F3F4F5',
+      border: '1px solid #E0E0E0',
+      textTransform: 'none',
+      margin: '0 0 10px 10px'
+    }
+  }));
+  const classes = useStyles();
+  const SHARE_BASE_URL = "https://git-contributor.com/";
+  const value = `<iframe style={{ width: "100%", height: "auto" , minWidth: "600px", minHeight: "1000px" }} src="${SHARE_BASE_URL}${params}" frameBorder="0"></iframe>`
```
Comment (guoqqqi) [inline src/components/DialogBox/index.jsx:42]:
```suggestion
  const embedCode = `<iframe style={{ width: "100%", height: "auto" , minWidth: "600px", minHeight: "1000px" }} src="${SHARE_BASE_URL}${params}" frameBorder="0"></iframe>`
```

File: src/components/DialogBox/index.jsx [hunk 16]
```diff
@@ -0,0 +1,155 @@
...
+  const [, setEmbedcopy] = useClipboard(value, { successDuration: 3000 })
+  const [, setMCopied] = useClipboard(`${SHARE_BASE_URL}${params}`, { successDuration: 3000 });
```
Comment (guoqqqi) [inline src/components/DialogBox/index.jsx:44]:
```suggestion
  const [, setCopy] = useClipboard(`${SHARE_BASE_URL}${params}`, { successDuration: 3000 });
```

File: src/components/DialogBox/index.jsx [hunk 17]
```diff
@@ -0,0 +1,155 @@
...
+  const [showNotice, setShowNotice] = React.useState(false);
+  const [showEmbedModal, setShowEmbedModal] = React.useState(false);
+  const [activeDate, setActiveDate] = React.useState("Link");
+
+  const SearchButton = () => (
+    <Button
+      className={classes.root}
+      value="Copy"
```
Comment (guoqqqi) [inline src/components/DialogBox/index.jsx:52]:
```suggestion
      value="embedCodeCopy"
```

File: src/components/DialogBox/index.jsx [hunk 18]
```diff
@@ -0,0 +1,155 @@
...
+    }
+  }));
+  const classes = useStyles();
+  const SHARE_BASE_URL = "https://git-contributor.com/";
+  const value = `<iframe style={{ width: "100%", height: "auto" , minWidth: "600px", minHeight: "1000px" }} src="${SHARE_BASE_URL}${params}" frameBorder="0"></iframe>`
+  const [, setEmbedcopy] = useClipboard(value, { successDuration: 3000 })
+  const [, setMCopied] = useClipboard(`${SHARE_BASE_URL}${params}`, { successDuration: 3000 });
+  const [showNotice, setShowNotice] = React.useState(false);
+  const [showEmbedModal, setShowEmbedModal] = React.useState(false);
+  const [activeDate, setActiveDate] = React.useState("Link");
```
Comment (guoqqqi) [inline src/components/DialogBox/index.jsx:47]: We usually write the React Hook at the top level and put some of the declared constants afterwards

File: src/components/contributor/index.js [hunk 19]
```diff
@@ -2,18 +2,23 @@ import React from "react";
...
 import { saveAs } from 'file-saver';
 
+import { DialogBox } from '../DialogBox'
 import { getMonths, getParameterByName, handleShareToTwitterClick } from "../../utils";
 import { generateDefaultOption } from "../../constants";
 import { fetchData, fetchMergeContributor } from "./service";
 import CustomizedDialogs, { MarkdownLink } from "../shareDialog";
 import { DEFAULT_COLOR } from "../../constants";
+import { Opacity } from "@material-ui/icons";
+
```
Comment (guoqqqi) [inline src/components/contributor/index.js:21]:
```suggestion
```

File: src/components/DialogBox/index.jsx [hunk 20]
```diff
@@ -0,0 +1,154 @@
...
+      backgroundColor: '#F3F4F5',
+      border: '1px solid #E0E0E0',
+      textTransform: 'none',
+      margin: '0 0 10px 10px'
+    }
+  }));
+  const classes = useStyles();
+  const [showNotice, setShowNotice] = React.useState(false);
+  const [showEmbedModal, setShowEmbedModal] = React.useState(false);
+  const [activeDate, setActiveDate] = React.useState("Link");
```
Comment (guoqqqi) [inline src/components/DialogBox/index.jsx:42]:
```suggestion
  const [showNotice, setShowNotice] = React.useState(false);
  const [showEmbedModal, setShowEmbedModal] = React.useState(false);
  const [activeDate, setActiveDate] = React.useState("Link");
  const useStyles = makeStyles(() => ({
    root: {
      backgroundColor: '#e53e3e',
      color: '#fff',
      border: 'none',
      padding: '0 10px',
      textTransform: 'none',
      margin: '0 0 10px 10px',
      lineHeight: '2.25rem',
      '&:hover': {
        backgroundColor: '#c53030'
      }
    },
    color: {
      color: '#3e434a',
      backgroundColor: '#F3F4F5',
      border: '1px solid #E0E0E0',
      textTransform: 'none',
      margin: '0 0 10px 10px'
    }
  }));
  const classes = useStyles();
```

File: src/components/shareDialog/index.js [hunk 21]
```diff
@@ -17,6 +17,7 @@ import SyntaxHighlighter from "react-syntax-highlighter/dist/esm/default-highlig
 import { a11yDark } from "react-syntax-highlighter/dist/esm/styles/hljs";
 import useClipboard from "react-use-clipboard";
 
+
```
Comment (guoqqqi) [inline src/components/shareDialog/index.js:20]:
```suggestion
```

**Comment by guoqqqi** [comment]
@SkyeYoung PTAL~

File: src/components/DialogBox/index.jsx [hunk 22]
```diff
@@ -0,0 +1,154 @@
...
+          }}
+        >
+          Link
+        </Button>
+        <Button
+          className={classes.root}
+          size="small"
+          value="Share on twtter"
+          startIcon={<svg stroke="currentColor" fill="currentColor" stroke-width="0" viewBox="0 0 512 512" height="0.9em" width="0.9em" xmlns="http://www.w3.org/2000/svg"><path width='1.8em' height='1.8em' d="M459.37 151.716c.325 4.548.325 9.097.325 13.645 0 138.72-105.583 298.558-298.558 298.558-59.452 0-114.68-17.219-161.137-47.106 8.447.974 16.568 1.299 25.34 1.299 49.055 0 94.213-16.568 130.274-44.832-46.132-.975-84.792-31.188-98.112-72.772 6.498.974 12.995 1.624 19.818 1.624 9.421 0 18.843-1.3 27.614-3.573-48.081-9.747-84.143-51.98-84.143-102.985v-1.299c13.969 7.797 30.214 12.67 47.431 13.319-28.264-18.843-46.781-51.005-46.781-87.391 0-19.492 5.197-37.36 14.294-52.954 51.655 63.675 129.3 105.258 216.365 109.807-1.624-7.797-2.599-15.918-2.599-24.04 0-57.828 46.782-104.934 104.934-104.934 30.213 0 57.502 12.67 76.67 33.137 23.715-4.548 46.456-13.32 66.599-25.34-7.798 24.366-24.366 44.833-46.132 57.827 21.117-2.273 41.584-8.122 60.426-16.243-14.292 20.791-32.161 39.308-52.628 54.253z"></path></svg>}
```
Comment (Baoyuantop) [inline src/components/DialogBox/index.jsx:122]: Save it in an SVG file?

**Comment by SkyeYoung** [comment]
### Why the borders are different, when I hovered？
image

### Darkening the btn may not be a good idea?
image

### Why don't these btns have hover state?
image

### The PR title need amend
It doesn't look like it's just a modified twitter button, and it's not just one feature.

**Comment by SkyeYoung** [comment]
image

Care about the paddings.

**Review by guoqqqi — APPROVED** [review]

**Pull request merged**